pkg/scheduler/schedule_one_test.go

```diff
@@ -1270,47 +1270,105 @@ func TestUpdatePod(t *testing.T) {
 
 func TestSelectHost(t *testing.T) {
 	tests := []struct {
-		name          string
-		list          []framework.NodePluginScores
-		possibleHosts sets.String
-		expectsErr    bool
+		name                                        string
+		list                                        []framework.NodePluginScores
+		numberOfNodeScoresToIncludeInScheduleResult int
+		possibleHosts                               sets.Set[string]
+		returnedNodeScoreListAssertFunc             func(got []framework.NodePluginScores) bool
```

Comment:
rather than a function, use "possibleNodeLists", and at the end compare got with the possible lists using cmp.diff to find at least one match.